ruby/test/syskit/test_syskit_plugin.rb

```diff
@@ -74,6 +74,30 @@
         assert_equal 'world_global', task.properties.world_frame
     end
 
+    it "does not attempt to use a producer's own output to configure it" do
```

Comment:
Just to be sure, if one of the self produced transforms is "a" => "b", both "a" => "b" and "b" => "a" would be registered? If so, shouldnt we also exclude "b" => "a" from the registered self produced transforms?

Reply:
Yes. This is handled by `transformation_chain`

I've added tests for this (and realized that the changes to transformation_chain were wrongly added to #10 instead of this PR ...)